mods/boats/init.lua

```diff
@@ -28,6 +28,29 @@ local function get_v(v)
 	return math.sqrt(v.x ^ 2 + v.z ^ 2)
 end
 
+
+local function after_detach(name, pos)
+	local player = minetest.get_player_by_name(name)
+	if player then
+		player:set_pos(pos)
+	end
+end
+
+
+local function after_attach(name)
+	local player = minetest.get_player_by_name(name)
+	if player then
+		player_api.set_animation(player, "sit" , 30)
+	end
+end
+
+
+local function after_remove(object)
+	if object then
```

Comment:
`object` will never be `nil`, because this reference is passed as a copy to `after_remove`.

Reply:
I agree, this is not needed.

Reply:
The reference is somehow set to `nil`: https://github.com/minetest/minetest_game/pull/2347#issuecomment-476965778